presto-raptor/src/main/java/com/facebook/presto/raptor/storage/OrcStorageManager.java

```diff
@@ -435,6 +459,10 @@ private ShardInfo createShardInfo(UUID shardUuid, OptionalInt bucketNumber, Path
         Path input = storageService.getStorageFile(shardUuid);
         Path output = storageService.getStagingFile(newShardUuid);
 
+        if (deltaDeleteEnabled.isPresent() && deltaDeleteEnabled.get()) {
```

Comment:
Not a big fan of checking a flag so deep inside the codebase with an `if`. Can we move it to some early phase? Ideally, `writeDeltaDeleteFile` and `rewriteFile` should be abstracted into two different rewriter impls of `FileRewriter`. I guess if that is the case, #13478 is not needed anymore.

Reply:
> Not a big fan of checking a flag so deep inside the codebase with an `if`. Can we move it to some early phase? Ideally, `writeDeltaDeleteFile` and `rewriteFile` should be abstracted into two different rewriter impls of `FileRewriter`. I guess if that is the case, #13478 is not needed anymore.

Great advice, change already made, thanks!